libraries/standard/mqtt/src/iot_mqtt_serialize.c

```diff
@@ -973,6 +1013,9 @@ static void _serializePublish( const IotMqttPublishInfo_t * pPublishInfo,
     /* The payload is placed after the packet identifier. */
     if( pPublishInfo->payloadLength > 0U )
     {
+        /* This memcpy intentionally copies bytes from a void * buffer into
```

Comment:
Similar to previous comment – we have `publishPacketSize` but we're not paying attention to it.  Let's compare against it to make sure we won't overrun after `pBuffer` on line 1019.  This makes the code a little safer and a little less scary for any future reviewers.  That will also allow you to delete 958.

Reply:
I'm not sure checking against `publishPacketSize` is necessary here since the buffer and its size are always checked whenever `_serializePublish` is called. Also, the buffer is written into several times before this line, so checking that `( ( ( size_t ) ( pBuffer - pPacket ) ) + pPublishInfo->payloadLength ) <= publishPacketSize` seems out of place here as it's not checked anywhere else in the function. Refactoring the function to make this check every time seems outside the scope of this PR.

Reply:
These are good points, I agree and don't think we should touch this for now.  At some point in the future however, we should think about an abstraction for working with these buffers rather than the direct approach used in this code.